app/models/manageiq/providers/openshift/container_manager/refresh_parser.rb

```diff
@@ -47,6 +52,71 @@ def parse_route(route)
                                                                 new_result[:namespace], get_service_name(route))
         new_result
       end
+
+      def parse_strings(ar_item, keys)
+        result = {}
+        keys.each do |key|
+          result.merge!(key => ar_item.try(key))
+        end
+        result
+      end
+
+      def parse_git_source(git_source)
+        parse_strings(git_source,
+                      [:uri, :ref, :httpproxy, :httpsproxy])
+      end
+
+      def parse_binary_source(binary_source)
+        binary_source ? binary_source.asFile : nil
+      end
+
+      def parse_source_secret(secret)
+        secret ? secret.name : nil
+      end
+
+      def parse_build_source(source_item)
+        {
+          :build_source_type => source_item.type,
+          :binary            => parse_binary_source(source_item.binary),
+          :dockerfile        => source_item.dockerfile,
+          :git               => parse_git_source(source_item.git).try(:uri),
```

Comment:
If only ```uri``` interests you here then I would do something like:
```Ruby
source_item.try(:git).try(:uri),
```

I think that having something like ```.try(:source_item).try(:git).try(:uri)``` isn't so bad.
what do you think @moolitayer ?